out_writeapi.go

```diff
@@ -160,6 +162,18 @@ func checkResponses(curr_ctx context.Context, currQueuePointer *[]*managedwriter
 	return 0
 }
 
+func sendRequest(ctx context.Context, data [][]byte, config **outputConfig) (*managedwriter.AppendResult, error) {
```

Comment:
Let's also move the 

if len(binaryData) > 0 

check to occur within this function, so it can be removed from both callers.